microservice/src/test/java/de/gsi/microservice/concepts/cmwlight/CmwLightClient.java

```diff
@@ -0,0 +1,512 @@
+    private static final AtomicInteger connectionId = new AtomicInteger(0); // global counter incremented for each connection
+    private final AtomicInteger channelId = new AtomicInteger(0); // connection local counter incremented for each channel
+    private final ZContext context = new ZContext();
+    private final ZMQ.Socket controlChannel;
+    private final AtomicReference<ConnectionState> connectionState = new AtomicReference<>(ConnectionState.DISCONNECTED);
+    private final String address;
+    private final IoClassSerialiser serialiser = new IoClassSerialiser(new FastByteBuffer(0), CmwLightSerialiser.class);
+
+    public ZMQ.Socket getSocket() {
+        return controlChannel;
+    }
+
+    public enum ConnectionState {
+        DISCONNECTED,
+        CONNECTING,
+        CONNECTED
+    }
+
+    // Contents of the first frame determine the type of the message
+    private static final byte SERVER_CONNECT_ACK = (byte) 0x01;
+    private static final byte SERVER_REP = (byte) 0x02;
+    private static final byte SERVER_HB = (byte) 0x03;
+    private static final byte CLIENT_CONNECT = (byte) 0x20;
+    private static final byte CLIENT_REQ = (byte) 0x21;
+    private static final byte CLIENT_HB = (byte) 0x22;
+
+    // Message Types in the descriptor (Last part of a message containing the type of each sub message)
+    private static final byte MT_HEADER = 0;
+    private static final byte MT_BODY = 1; //
+    private static final byte MT_BODY_DATA_CONTEXT = 2;
+    private static final byte MT_BODY_REQUEST_CONTEXT = 3;
+    private static final byte MT_BODY_EXCEPTION = 4;
+
+    // Field names for the Request Header
+    public static final String EVENT_TYPE_TAG = "eventType";
+    public static final String MESSAGE_TAG = "message";
+    public static final String ID_TAG = "0";
+    public static final String DEVICE_NAME_TAG = "1";
+    public static final String REQ_TYPE_TAG = "2";
+    public static final String OPTIONS_TAG = "3";
+    public static final String CYCLE_NAME_TAG = "4";
+    public static final String ACQ_STAMP_TAG = "5";
+    public static final String CYCLE_STAMP_TAG = "6";
+    public static final String UPDATE_TYPE_TAG = "7";
+    public static final String SELECTOR_TAG = "8";
+    public static final String CLIENT_INFO_TAG = "9";
+    public static final String NOTIFICATION_ID_TAG = "a";
+    public static final String SOURCE_ID_TAG = "b";
+    public static final String FILTERS_TAG = "c";
+    public static final String DATA_TAG = "x";
+    public static final String SESSION_ID_TAG = "d";
+    public static final String SESSION_BODY_TAG = "e";
+    public static final String PROPERTY_NAME_TAG = "f";
+
+    // request type used in request header REQ_TYPE_TAG
+    public static final byte RT_GET = 0;
+    public static final byte RT_SET = 1;
+    public static final byte RT_CONNECT = 2;
+    public static final byte RT_REPLY = 3;
+    public static final byte RT_EXCEPTION = 4;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/GSI-CS-CO/chart-fx/pullRequest?prid=6417799)